pallets/domains/src/lib.rs

```diff
@@ -411,58 +362,81 @@ pub mod pallet {
             }
 
             let expires_at = expires_in.saturating_add(System::<T>::block_number());
-            let domain_meta = DomainMeta::new(
-                expires_at,
-                owner.clone(),
-                content,
-                deposit,
-            );
+            let domain_meta = DomainMeta::new(expires_at, owner.clone(), deposit);
 
             // TODO: withdraw balance when it will be possible to purchase domains.
 
             RegisteredDomains::<T>::insert(domain_lc.clone(), domain_meta);
-            DomainsByOwner::<T>::mutate(
-                &owner, |domains| {
-                    domains.try_push(domain_lc.clone()).expect("qed; too many domains per account")
-                }
-            );
+            DomainsByOwner::<T>::mutate(&owner, |domains| {
+                domains.try_push(domain_lc.clone()).expect("qed; too many domains per account")
+            });
 
             Self::deposit_event(Event::DomainRegistered { who: owner, domain: full_domain });
             Ok(())
         }
 
-        fn do_set_inner_value(
-            full_domain: DomainName<T>,
-            new_value_opt: Option<InnerValueOf<T>>,
+        fn do_set_record(
+            domain: DomainName<T>,
+            key: RecordKey<T>,
+            value_opt: Option<RecordValue<T>>,
             check_ownership: Option<T::AccountId>,
+            reserve_deposit: bool,
         ) -> DispatchResult {
-            let domain_lc = Self::lower_domain_then_bound(&full_domain);
+            let domain_lc = Self::lower_domain_then_bound(&domain);
             let meta = Self::require_domain(domain_lc.clone())?;
+            let owner = meta.owner.clone();
 
             if let Some(should_be_owner) = check_ownership {
                 Self::ensure_allowed_to_update_domain(&meta, &should_be_owner)?;
             }
 
-            ensure!(meta.inner_value != new_value_opt, Error::<T>::InnerValueNotChanged);
-
-            if let Some(old_value) = &meta.inner_value {
-                DomainByInnerValue::<T>::remove(&meta.owner, old_value);
-            }
+            let (old_depositor, old_deposit): (<T as frame_system::pallet::Config>::AccountId, BalanceOf<T>) = if let Some(value_with_deposit_info) =
+                DomainRecords::<T>::get(domain_lc.clone(), key.clone())
+            {
+                (value_with_deposit_info.1, value_with_deposit_info.2)
```

Comment:
I'd better use bindings instead of indexes, because then it'll be more clear, what data exactly we're getting from the tuple.

Reply:
You mean we should use a struct for this?